homeassistant/components/nightscout/__init__.py

```diff
@@ -0,0 +1,67 @@
+"""The Nightscout integration."""
+import asyncio
+import logging
+
+from py_nightscout import Api as NightscoutAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import SLOW_UPDATE_WARNING
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["sensor"]
+_API_TIMEOUT = SLOW_UPDATE_WARNING - 1
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Nightscout component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Nightscout from a config entry."""
+    server_url = entry.data[CONF_URL]
+
+    api = NightscoutAPI(server_url)
+    status = await api.get_server_status()
```

Comment:
can this throw exceptions?

Reply:
Totally missed that! I've added the exception handling and a test for it.